src/dials/algorithms/correlation/analysis.py

```diff
@@ -182,7 +198,11 @@ def calculate_matrices(self):
         self.cosym_analysis._intialise_target()
 
         # Cosym proceedures to calculate the cos-angle matrix
-        self.cosym_analysis._determine_dimensions()
+        if self.params.dimensions is Auto:
+            self.cosym_analysis._determine_dimensions(
+                self.cosym_analysis.number_of_datasets,
```

Comment:
```suggestion
                len(self.unmerged_datasets),
```
My changeset removed the `number_of_datasets` variable, so this line needs updating

Reply:
Further, we probably want to have a phil parameter of the maximum number of dimensions to test (default 20?, 50?) otherwise it will become very slow for processing >>100s of datasets

Reply:
Added in a phil parameter for max dimensions - set default as 50 as important to have good trend for analysis, and put the smaller number into self.cosym_analysis._determine_dimensions